R/page.R

```diff
@@ -297,12 +297,14 @@ page_sidebar <- function(
   )
 }
 
-page_main_container <- function(...) {
-  as_fill_carrier(
+page_main_container <- function(..., fillable = TRUE) {
+  htmltools::bindFillRole(
     tags$main(
       class = "bslib-page-main bslib-gap-spacing",
       ...
-    )
+    ),
+    item = TRUE,
```

Comment:
I'm probably missing something here, but if the parent of `bslib-page-main` is not `fillable`, why does it matter if this is a fill item?

```suggestion
    item = FALSE,
```

Reply:
Maybe not? But I know for sure that, without having to consider parent containers, we want to break fillability for children by making sure the container isn't a fill container. We could probably also set `item = fillable`, but then we have to consider all of the parent layouts. `page_main_container()` isn't just used in `page_sidebar()`, it's also used in `navbar_page()` with a sidebar where the DOM structure is quite different from `page_sidebar()`.

I know for certain that removing `html-fill-container` does what we want it to do in this case and that that's correct in all layouts. That's also why I used `.html-fill-container` for `min-height`/`min-width` above. My general take is that `.html-fill-item` is upward-indicating, as in it makes the child filling when the parent context is fillable. `.html-fill-container` is downward-facing and communicates intent downward to its children.

My take is that we could _probably_ either make `.bslib-page-main` a fill carrier when `fillable = TRUE` or add no filling properties when `fillable = FALSE`, but we could _certainly_ work with the fill container property. So this choice mostly came down to 1) proving we want `container = fillable` is easy; 2) proving we want `item = fillable` requires enumerating all the layouts and making sure that's right.

Reply:
Ah ok, I was missing that `navbar_page()` context. Although I'll probably find the code confusing to revisit, I like the careful approach your taking. Maybe consider linking to your comment here in the code?

Reply:
Well, in the mean time I convinced myself that we should go wholesale `as_fill_carrier()` when `fillable=TRUE` and just wrap things in the `<main>` tag otherwise.

I think the reason I used `as_fill_carrier()` was in case we use `page_main_container()` in other situations; the fact that that also adds `display: flex` was the unintended side-effect.